### PR TITLE
pg_cron (new formula)

### PR DESCRIPTION
**Pull request by mtesch-um**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/pg_cron.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  homepage "https://github.com/citusdata/pg_cron"
+  url "https://github.com/citusdata/pg_cron/archive/refs/tags/v1.4.1.tar.gz"
+  sha256 "f50792baa7105f7e5526ec0dca6d2328e4290131247c305cc84628cd864e9da2"
+  license "PostgreSQL"
+
+  depends_on "postgresql"
+
+  def install
+    system "make"
+    (prefix/"lib/postgresql").install "pg_cron.so"
```
Comment (SMillerDev) [inline Formula/pg_cron.rb:12]:
```suggestion
    (lib/"postgresql").install "pg_cron.so"
```

File: Formula/pg_cron.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    (prefix/"share/postgresql/extension").install "pg_cron--1.0.sql"
+    (prefix/"share/postgresql/extension").install "pg_cron--1.0--1.1.sql"
+    (prefix/"share/postgresql/extension").install "pg_cron--1.1--1.2.sql"
+    (prefix/"share/postgresql/extension").install "pg_cron--1.2--1.3.sql"
+    (prefix/"share/postgresql/extension").install "pg_cron--1.3--1.4.sql"
```
Comment (SMillerDev) [inline Formula/pg_cron.rb:17]:
```suggestion
    (share/"postgresql/extension").install Dir["pg_cron--*.sql"]
```

File: Formula/pg_cron.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    (prefix/"share/postgresql/extension").install "pg_cron.control"
```
Comment (SMillerDev) [inline Formula/pg_cron.rb:18]:
```suggestion
    (share/"postgresql/extension").install "pg_cron.control"
```

File: Formula/pg_cron.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  end
+
+  def caveats
```
Comment (SMillerDev) [inline Formula/pg_cron.rb:21]: This should be in the upstream documentation, it's not homebrew specific

File: Formula/pg_cron.rb [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  end
+
+  test do
+    # Testing steps:
+    # - create new temporary postgres database
+    system "pg_ctl", "initdb", "-D", testpath/"test"
+
+    # - enable pg_cron in temporary database
+    (testpath/"test/postgresql.conf").write("\nshared_preload_libraries = 'pg_cron'\n", mode: "a+")
+    (testpath/"test/postgresql.conf").write("\nport = 5562\n", mode: "a+")
```
Comment (SMillerDev) [inline Formula/pg_cron.rb:41]:
```suggestion
    port = free_port
    # - enable pg_cron in temporary database
    (testpath/"test/postgresql.conf").write("\nshared_preload_libraries = 'pg_cron'\n", mode: "a+")
    (testpath/"test/postgresql.conf").write("\nport =#{port}\n", mode: "a+")
```

File: Formula/pg_cron.rb [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+    # - restart temporary postgres
+    system "pg_ctl", "start", "-D", testpath/"test", "-l", testpath/"log"
+
+    # - run "CREATE EXTENSION pg_cron;" in temp database
+    system "psql", "-p", "5562", "-c", "CREATE EXTENSION pg_cron;", "postgres"
```
Comment (SMillerDev) [inline Formula/pg_cron.rb:47]:
```suggestion
    system "psql", "-p", port.to_s, "-c", "CREATE EXTENSION pg_cron;", "postgres"
```

File: Formula/pg_cron.rb [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+      Restart and then create the extension in the database:
+        -- run as superuser:
+        CREATE EXTENSION pg_cron;
+      For more details, read:
+        https://github.com/citusdata/pg_cron
+    EOS
+  end
+
+  test do
```
Comment (mtesch-um) [inline Formula/pg_cron.rb:30]:
```suggestion
  end

  test do
```

File: Formula/pg_cron.rb [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    # Testing steps:
+    # - create new temporary postgres database
+    system "pg_ctl", "initdb", "-D", testpath/"test"
+
+    port = free_port
+    # - enable pg_cron in temporary database
+    (testpath/"test/postgresql.conf").write("\nshared_preload_libraries = 'pg_cron'\n", mode: "a+")
+    (testpath/"test/postgresql.conf").write("\nport =#{port}\n", mode: "a+")
```
Comment (carlocab) [inline Formula/pg_cron.rb:25]:
```suggestion
    (testpath/"test/postgresql.conf").write <<~EOS, mode: "a+"
   
      shared_preload_libraries = 'pg_cron'
      port = #{port}
    EOS
```

**Review by SMillerDev — DISMISSED** [review]

File: Formula/pg_cron.rb [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+  test do
+    # Testing steps:
+    # - create new temporary postgres database
+    system "pg_ctl", "initdb", "-D", testpath/"test"
+
+    port = free_port
+    # - enable pg_cron in temporary database
+    (testpath/"test/postgresql.conf").write <<~EOS, mode: "a+"
+   
+      shared_preload_libraries = 'pg_cron'
```
Comment (carlocab) [inline Formula/pg_cron.rb:26]:
```suggestion
    (testpath/"test/postgresql.conf").write <<~EOS, mode: "a+"

      shared_preload_libraries = 'pg_cron'
```
Oops, I forgot GitHub adds a bunch of whitespace automatically.

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@mtesch-um, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**